worlds/earthbound/modules/boss_shuffle.py

```diff
@@ -0,0 +1,411 @@
+        "Giygas": BossData(0x0172, 0xEEF095, 0xEEF095, 0x01DD, 0xDC, 0x49)
+    }
+
+    if world.options.skip_prayer_sequences:
+        # Boss shuffle sprites needs to apply to the skip prayer cleanup too
+        world.boss_slots["Giygas"].sprite_addrs.append(0x07B9AC)
+        world.boss_slots["Heavily Armed Pokey"].sprite_addrs.append(0x07B9A7)
+
+    # mole/rat text
+    # todo; Giygas sprites/text
+
+    world.boss_slot_order = world.boss_list.copy()
+    if type(world.options.boss_shuffle.value) == str:
+        boss_plando = world.options.boss_shuffle.value.split(";")
+        shuffle_result = boss_plando.pop()
+    else:
+        boss_plando = []
+        shuffle_result = world.options.boss_shuffle.value
+
+    if shuffle_result == "true" or shuffle_result == 1:
+        world.random.shuffle(world.boss_list)
+
+        if not world.options.decouple_diamond_dog:
+            world.boss_list.remove("Diamond Dog")
+            insert_index = 28 if not world.options.boss_shuffle_add_giygas else 27
+            world.boss_list.insert(insert_index, "Diamond Dog")
+
+        if not world.options.boss_shuffle_add_giygas:
+            world.boss_list.remove("Giygas")
+            world.boss_list.insert(29, "Giygas")
+
+    if world.options.safe_final_boss:
+        while world.boss_list[25] in hard_final_bosses:
+            i = world.random.randrange(len(world.boss_list))
+            if (world.boss_list[i] == "Diamond Dog" and not world.options.decouple_diamond_dog) or (
+                world.boss_list[i] == "Giygas" and not world.options.boss_shuffle_add_giygas
+            ):
+                continue
+            world.boss_list[25], world.boss_list[i] = world.boss_list[i], world.boss_list[25]
+
+    did_plando_diam_dog = False
+    diamond_dog_plando_slot = None
+
+    for item in boss_plando:
+        boss_block = item.split("-")
+        boss = boss_block[0].title()
+        slot = boss_block[1].title()
+        if boss in boss_typo_key:
+            boss = boss_typo_key[boss]
+
+        if slot in boss_typo_key:
+            slot = boss_typo_key[boss]
+
+        if slot == "Diamond Dog":
+            did_plando_diam_dog = True
+            diamond_dog_plando_slot = boss
+
+        old_index = world.boss_list.index(boss)
+        new_index = world.boss_list.index(slot)
+
+        world.boss_list[old_index] = slot
+        world.boss_list[new_index] = boss
+
+    if world.boss_list[25] == "Carbon Dog" and world.boss_list[27] in banned_transformations:
+        if did_plando_diam_dog:
+            warning(f"""Unable to plando {diamond_dog_plando_slot} for {world.multiworld.get_player_name(world.player)}'s EarthBound world.
+This boss cannot be placed onto Diamond Dog's slot if Carbon Dog is on Heavily Armed Pokey's slot.
+This message is likely the result of randomization and can be safely ignored.""")  # Why is this spacing the only way to get the message to render legibly
```

Comment:
Yeah, multiline strings look kinda ugly.

The only alternative would be sth like this:

```py
warning(
    f"Unable to plando {diamond_dog_plando_slot} for {world.player_name}'s EarthBound world.\n"
    "This boss cannot be placed onto Diamond Dog's slot if Carbon Dog is on Heavily Armed Pokey's slot.\n"
    "This message is likely the result of randomization and can be safely ignored."
)
```

Reply:
Anyway, feel free to just resolve this, I just wanted to answer your question :)